app.py

```diff
@@ -135,28 +135,16 @@ def render_and_submit_path(self, path, fields, first_frame, last_frame, sg_publi
         if version_template:
             version_name = version_template.apply_fields(fields)
 
-        # create a dict with fields and some additional information about the
-        # current entity to replace tokens with the same name present in the
-        # review submission nuke script
-        replace_data = copy.deepcopy(fields)
+        # Get head_in, tail_out to pass to submitter
         if self.context.entity:
             sg_entity_type = self.context.entity["type"]
             sg_filters = [["id", "is", self.context.entity["id"]]]
-
-            sg_fields = ["smart_cut_summary_display",
-                         "sg_client_name",
-                         "sg_lens___primary",
-                         "sg_head_in",
+            sg_fields = ["sg_head_in",
                          "sg_tail_out"]
-            replace_data.update(self.shotgun.find_one(sg_entity_type,
-                                                      filters=sg_filters,
-                                                      fields=sg_fields))
-
-        replace_data["first_frame"] = first_frame
-        replace_data["last_frame"] = last_frame
-        replace_data["description"] = comment
-        # TODO: publisher.util.get_publish_name() does this much better
-        replace_data["file_base_name"] = os.path.basename(path).split('.')[0]
+            fields.update(self.shotgun.find_one(sg_entity_type,
+                                                filters=sg_filters,
+                                                fields=sg_fields))
+        fields["description"] = comment
```

Comment:
Moved as much replace data creation as possible to hook, but passing what information is already available here in fields.